### ISSUE TITLE
FreeBSD 10.1: unknown type u_{int,short,char} errors in misc.c

### ISSUE THREAD

**Issue by yggdr** [vergoh/vnstat#27]
I see errors like this 

    In file included from common.h:18:0,
                     from misc.c:2:
    /usr/include/sys/file.h:209:2: error: unknown type name 'u_int'
      u_int xf_flag; /* flags (see fcntl.h) */

in `gmake all`, even with the CPATH and LIBRARY_PATH variables set. I'm not a C programmer, so what other information do you need?

**Comment by vergoh** [comment]
Which version of vnStat did you get this error with? Does the FreeBSD ports provided version compile any better?

A quick lookup regarding that error suggests it's related to the _XOPEN_SOURCE defintion I've used in misc.c and how the FreeBSD headers react to the definition. I found few suggestions how to deal with this sort of error but I'll have to validate the result once I've setup a suitable environment for testing it out.
